### ISSUE TITLE
Invalid constraint definition error without indication of where the error is

### ISSUE THREAD

**Issue by fmoor** [edgedb/edgedb#2305]
This error doesn't give the user much help locating the error in their file. It would be nice to have a line number and column.

```bash
$ edgedb -I workout create-migration
ERROR: InvalidConstraintDefinitionError: constraints cannot contain paths with more than one hop
edgedb error: cannot proceed until .esdl files are fixed
```

schema.esdl
```
module default {
	type Exercise {
		required property name -> str {
			constraint exclusive;
		};
	}
	type Set_ {
		link workout := .<sets[IS Workout];
		required link exercise -> Exercise;
		required property index -> int16 {
			constraint min_value(0);
		};
		required property weight -> float32;
		required property reps -> int16;
		required property completed -> bool {
			default := false;
		};

		constraint exclusive on ( (.index, .workout) );
	}
	type Workout {
		multi link sets -> Set_ {
			constraint exclusive;
		}
		required property started_at -> datetime {
			constraint exclusive;
			default := datetime_current();
		}
	}
};
```

- EdgeDB Version: package:1.0b2.dev5478+g25318fe0e-2021022200~focal.nightly
- EdgeDB CLI Version: edgedb-cli 1.0.0-beta.1
- OS Version: Linux 5.8.0-44-generic edgedb/edgedb-cli#50~20.04.1-Ubuntu SMP Wed Feb 10 21:07:30 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

**Comment by tailhook** [comment]
I believe this is server issue. Let me know if not.

**Comment by msullivan** [comment]
I added this error message pretty recently and it is *supposed* to come with error context, so I'll take a look
